src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Deployments/RemoveAzureDeploymentCmdlet.cs

```diff
@@ -0,0 +1,76 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Components;
+using Microsoft.Azure.Commands.ResourceManager.Cmdlets.SdkModels;
+using ProjectResources = Microsoft.Azure.Commands.ResourceManager.Cmdlets.Properties.Resources;
+
+namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation
+{
+    /// <summary>
+    /// Deletes a deployment.
+    /// </summary>
+    [Cmdlet(VerbsCommon.Remove, "AzureRmDeployment", SupportsShouldProcess = true,
+        DefaultParameterSetName = RemoveAzureDeploymentCmdlet.DeploymentNameParameterSet), OutputType(typeof(bool))]
+    public class RemoveAzureDeploymentCmdlet : ResourceManagerCmdletBase
+    {
+        /// <summary>
+        /// The deployment Id parameter set.
+        /// </summary>
+        internal const string DeploymentIdParameterSet = "RemoveByDeploymentId";
+
+        /// <summary>
+        /// The deployment name parameter set.
+        /// </summary>
+        internal const string DeploymentNameParameterSet = "RemoveByDeploymentName";
+
+        /// <summary>
+        /// The input object parameter set.
+        /// </summary>
+        internal const string InputObjectParameterSet = "RemoveByInputObject";
+
+        [Alias("DeploymentName")]
+        [Parameter(Position = 0, ParameterSetName = RemoveAzureDeploymentCmdlet.DeploymentNameParameterSet, Mandatory = true,
+            ValueFromPipelineByPropertyName = true, HelpMessage = "The name of the deployment.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Alias("DeploymentId", "ResourceId")]
+        [Parameter(ParameterSetName = RemoveAzureDeploymentCmdlet.DeploymentIdParameterSet, Mandatory = true,
+            HelpMessage = "The fully qualified resource Id of the deployment. example: /subscriptions/{subId}/providers/Microsoft.Resources/deployments/{deploymentName}")]
+        [ValidateNotNullOrEmpty]
+        public string Id { get; set; }
+
+        [Parameter(ParameterSetName = RemoveAzureDeploymentCmdlet.InputObjectParameterSet, Mandatory = true, 
+            ValueFromPipeline = true, HelpMessage = "The deployment object.")]
+        public PSDeployment InputObject { get; set; }
+
```

Comment:
You may want to add -AsJob switch to the cmdlet.

Reply:
We don't have this parameter in Remove-AzureRmResourceGroupDeployment. I think we can leave it. We can always add it later if people ask for it.

Reply:
@Tiano2017 is the underlying operation long-running? If so, you might as well add the `-AsJob` parameter to allow users to run this cmdlet in the background (at no cost 😉)